README.md

```diff
@@ -1,42 +1,62 @@
+- [Testing](https://docs.flutter.dev/testing)-ready (unit, integration, and widget testing), [production and deployment](https://docs.flutter.dev/deployment)-ready (to Firebase, Play Store, TestFlight, and AppStore).
+- Built-in [GitHub templates & CI/CD workflows](bricks/template/__brick__/%7B%7Bproject_name.snakeCase()%7D%7D/.github) integrated with Github Action to perform static code analysis, test, build and deploy app builds to app distribution services or app stores.
 
-`git clone git@github.com:nimblehq/flutter-templates.git`
+### Optional (enable by [generator command parameters](#set-up-a-new-project))
 
-## Documentation
-
-Check out the [Wiki](https://github.com/nimblehq/flutter-templates/wiki) page to access the full documentation.
+- Specify the default [JSON field renaming format with json_serializable](https://pub.dev/packages/json_serializable#build-configuration) between support values: none, kebab, snake, and pascal.
+- Integrate permissions requesting & checking with [permission_handler](https://pub.dev/packages/permission_handler) and its necessary basic setup.
+- [Code coverage integration](bricks/template/__brick__/%7B%7Bproject_name.snakeCase()%7D%7D/codecov.yml) by `Codecov`.
 
 ## Use the template
 
-### Setup a new project
+### Prerequisites
 
 Before using the template, ensure that you have installed the following prerequisites on your system:
 
 - Flutter 3.10.5
 - [Mason CLI](https://pub.dev/packages/mason_cli) 0.1.0-dev.44
 
+### Set up a new project
+
 Follow these steps to set up a new project from the template:
 
-- Fetch all required bricks by running the command:
+1. Clone or download this repository to your local machine.
+
+2. Fetch all required bricks by running the command:
+
+    `$ mason get`
 
-  `$ mason get`
+3. Generate the new project by running the following command with [supported arguments](bricks/template/brick.yaml):
 
-- Generate the new project by running the following command:
+    `$ mason make template` and then provide necessary inputs.
 
-  `$ mason make template`
+    or
 
-  > You can find the detailed information on `make` command options and usage in the [Mason documentation](https://github.com/felangel/mason/tree/master/packages/mason_cli#overview).
+    `$ mason make template -c mason-config.json` to provide necessary inputs via a JSON file, e.g., [mason-config.json](mason-config.json) for the sample project.
 
-- Once the project is generated at `/{project_name}`, please refer to the [Getting Started](https://github.com/nimblehq/flutter-templates/tree/develop/bricks/template/__brick__/%7B%7Bproject_name.snakeCase()%7D%7D#getting-started) documentation to make it ready for development.
+    Once the project is generated at `/{project_name}`, please refer to the [Getting Started](bricks/template/__brick__/%7B%7Bproject_name.snakeCase()%7D%7D#getting-started) documentation to make it ready for development.
```

Comment:
I think this should be step 4 or at the same level as step 3 (no indentation)

Reply:
Fixed in https://github.com/nimblehq/flutter-templates/pull/238/commits/ac20fbcbfa9e2d076ed0152921fb6417c1b7553b